src/services/codeFixProvider.ts

```diff
@@ -65,11 +81,15 @@ namespace ts.codefix {
         return createCombinedCodeActions(changes, commands.length === 0 ? undefined : commands);
     }
 
-    export function eachDiagnostic({ program, sourceFile, cancellationToken }: CodeFixAllContext, errorCodes: readonly number[], cb: (diag: DiagnosticWithLocation) => void): void {
-        for (const diag of program.getSemanticDiagnostics(sourceFile, cancellationToken).concat(computeSuggestionDiagnostics(sourceFile, program, cancellationToken))) {
+    export function eachDiagnostic(context: CodeFixAllContext, errorCodes: readonly number[], cb: (diag: DiagnosticWithLocation) => void): void {
+        for (const diag of getDiagnostics(context)) {
             if (contains(errorCodes, diag.code)) {
                 cb(diag as DiagnosticWithLocation);
             }
         }
     }
+
+    function getDiagnostics({ program, sourceFile, cancellationToken }: CodeFixContextBase) {
+        return program.getSemanticDiagnostics(sourceFile, cancellationToken).concat(computeSuggestionDiagnostics(sourceFile, program, cancellationToken));

```

Comment:
Does this do work or just retrieve the diagnostics from a cache?  Are syntactic diagnostics implicitly included in one of these categories?

Reply:
It does work if necessary, but the way this flow goes, the additional time I’m calling it will always hit the cache.

Reply:
That was my guess.  Thanks!

Reply:
That was my guess.  Thanks!